### PR TITLE
Share button state from mousekey to pointing_device

### PR DESCRIPTION
**Pull request by manna-harbour**
## Description

If you enable mousekey (mouse keys) with pointing_device (pointing device) and add button keycodes to your keymap, you'll get a click from mousekey but drags with pointing_device movement won't work as button state needs to be sent with the move. If you handle the mousekey button keycodes yourself in process_record_user() to update pointing_device button state you'll break drags using mousekey movement. If you handle mousekey movement keycodes too then you've just re-implemented mousekey.

This change shares button state from mousekey to pointing_device to support simultaneous use of mousekey and pointing_device, including mouse drags with mousekey buttons and mousekey or pointing_device movement.

To use, enable mousekey and pointing_device and add button keycodes to keymap.  No extra code is required. 

See also #9124.

## Types of Changes

- [x] Core
- [x] Enhancement/optimization
- [x] Documentation

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by manna-harbour** [comment]
Rebased onto force-pushed `develop`.

**Comment by manna-harbour** [comment]
The following users have keymaps with pointing_device and button keycodes and might like to try this: @drashna, @vlukash, @alairock, @dr-BEat, @Snipeye.

**Comment by manna-harbour** [comment]
Rebased after force-pushes to `develop`.

File: tmk_core/common/action.c [hunk 1]
```diff
@@ -220,6 +224,19 @@ void process_record_handler(keyrecord_t *record) {
     process_action(record, action);
 }
 
+#if defined PS2_MOUSE_ENABLE || defined POINTING_DEVICE_ENABLE
```
Comment (tzarc) [inline tmk_core/common/action.c:227]: Just for consistency with the rest of qmk (fully aware that these are already this exact format in other places of the file -- don't want to perpetuate it with new code):
```suggestion
#if defined(PS2_MOUSE_ENABLE) || defined(POINTING_DEVICE_ENABLE)
```

File: tmk_core/common/action.c [hunk 2]
```diff
@@ -404,15 +421,23 @@ void process_action(keyrecord_t *record, action_t action) {
             if (event.pressed) {
                 mousekey_on(action.key.code);
                 switch (action.key.code) {
-#    ifdef PS2_MOUSE_ENABLE
+#    if defined PS2_MOUSE_ENABLE || defined POINTING_DEVICE_ENABLE
```
Comment (tzarc) [inline tmk_core/common/action.c:424]: Again, consistency:
```suggestion
#    if defined(PS2_MOUSE_ENABLE) || defined(POINTING_DEVICE_ENABLE)
```

File: tmk_core/common/action.c [hunk 3]
```diff
@@ -422,15 +447,23 @@ void process_action(keyrecord_t *record, action_t action) {
             } else {
                 mousekey_off(action.key.code);
                 switch (action.key.code) {
-#    ifdef PS2_MOUSE_ENABLE
+#    if defined PS2_MOUSE_ENABLE || defined POINTING_DEVICE_ENABLE
```
Comment (tzarc) [inline tmk_core/common/action.c:450]: Just for consistency:
```suggestion
#    if defined(PS2_MOUSE_ENABLE) || defined(POINTING_DEVICE_ENABLE)
```

**Review by tzarc — CHANGES_REQUESTED** [review]
Just a couple of minor consistency changes.

**Comment by manna-harbour** [comment]
> Just a couple of minor consistency changes.

Thanks!

**Comment by manna-harbour** [comment]
Squashed.

**Comment by manna-harbour** [comment]
>￼ manna-harbour requested review from tzarc and removed request for qmk/collaborators

@tzarc Woops, sorry, wrong button!

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Comment by manna-harbour** [comment]
Thanks everyone!

**Pull request merged**